### PR TITLE
SONARJAVA-2022 Fix S2187 to handle 'cucumber' runner

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/test/files/checks/NoTestInTestClassCheck.java [hunk 1]
```diff
@@ -87,7 +88,13 @@ public void test2() {
 public class MyNewTest { // should not raise an issue
 }
 @org.junit.runner.RunWith(Enclosed.class)
-public class MyNewTest2 { // no issue
+public class MyNew2Test { // no issue
+}
+@org.junit.runner.RunWith(cucumber.api.junit.Cucumber.class)
```
Comment (vilchik-elena) [inline java-checks/src/test/files/checks/NoTestInTestClassCheck.java]: @m-g-sonar Not sure it's relevant but what about testing `@RunWith...` (without full name)?

File: java-checks/src/test/files/checks/NoTestInTestClassCheck.java [hunk 2]
```diff
@@ -87,7 +88,13 @@ public void test2() {
...
+public class MyCucumberTest { // should not raise an issue
+}
+@org.junit.runner.RunWith(Cucumber.class)
```
Comment (vilchik-elena) [inline java-checks/src/test/files/checks/NoTestInTestClassCheck.java]: @m-g-sonar You don't test unknown runners

File: java-checks/src/main/java/org/sonar/java/checks/NoTestInTestClassCheck.java [hunk 3]
```diff
@@ -99,16 +100,19 @@ private void checkJunit3TestClass(IdentifierTree className, Iterable<Symbol> mem
   }
 
   private void checkJunit4AndAboveTestClass(IdentifierTree className, JavaSymbol.TypeJavaSymbol symbol, Iterable<Symbol> members) {
-    if (symbol.name().endsWith("Test") && !runWithEnclosedRunner(symbol)) {
+    if (symbol.name().endsWith("Test") && !runWithRunner(symbol)) {
```
Comment (vilchik-elena) [inline java-checks/src/main/java/org/sonar/java/checks/NoTestInTestClassCheck.java]: @m-g-sonar `runWithRunner` this name is a bit funny :)

File: java-checks/src/test/files/checks/NoTestInTestClassCheck.java (hunk 2 above)
Comment (m-g-sonar) [inline java-checks/src/test/files/checks/NoTestInTestClassCheck.java]: Indeed @vilchik-elena we are currently specifically targeting `Cucumber.class` and `Enclosed.class`

**Pull request merged**